src/book.py

```diff
@@ -0,0 +1,224 @@
+from nltk.corpus import stopwords
+from nltk import FreqDist, word_tokenize, sent_tokenize, WordNetLemmatizer
+import re
+import random
+
+
+class BookText():
+    """A class for reading and manipulating texts"""
+
+    def __init__(self, filepath=None, rawtext=None, encoding='utf-8',
+                 clean=False, author=None, title=None, meta=None):
+        """Constructor for BookText
+
+        parameters:
+            **NB** one of filepath or rawtext must be specified
+
+            filepath (None): filepath to a text file to read
+            rawtext (None): a raw text string
+            encoding ("utf-8"): the encoding to use
+            clean (False): whether to clean the text on initializing
+            author (None): if not specified, inferred from text
+            title (None): if not specified, inferred from text
+        """
+        if not (filepath or rawtext):
+            raise ValueError('Must specify one of filepath or rawtext')
+        if filepath is not None:
```

Comment:
I suggest we stick with a general `IOError` (or maybe it's `OSError` now), which includes `FileNotFoundError`, `IsADirectoryError`, `PermissionError` and others. So,

My more general question is: what behavior do you want if an exception is found? We could create a BookText object empty text and None for all other fields, but I'm not sure that's helpful. It may actually be best if we let an error be raised so we know what the issue is.

Reply:
I think that there are some errors that are best handled by the functions that raise them. For example, if you give a bad filepath to pd.read_csv(), something else is spitting out the error. I agree with Elaad than if we're not going to do anything with the error then we should leave it alone.

Reply:
@NikhilTilak are you okay with leaving it as is?